src/queue/BaseBatchedJob.php

```diff
@@ -138,11 +143,18 @@ public function execute($queue): void
             $this->itemOffset++;
             $i++;
 
-            // Make sure we're not getting uncomfortably close to the memory limit, every 10 items
-            if ($startMemory !== null && $i % 10 === 0) {
-                $memory = memory_get_usage();
-                $avgMemory = ($memory - $startMemory) / $i;
-                if ($memory + ($avgMemory * 15) > $memoryLimit) {
+            // Make sure we're not getting uncomfortably close to the memory limit or TTL, every 10 items
+            if ($i % 10 === 0) {
```

Comment:
Why not just do this every time?

Reply:
I was just integrating it with the existing code, but we probably should.
Why bother for memory? Is `memory_get_usage` a taxing op?

Reply:
> Why bother for memory? Is `memory_get_usage` a taxing op?

Not that I know of.

Reply:
K, killed it for both 🤷‍♂️